### ISSUE TITLE
Associate *.ipynb files with IPython Notebook on Windows

### ISSUE THREAD

**Issue by petebachant** [ContinuumIO/anaconda-issues#267]
It would be helpful if when a user opens an *.ipynb file from Windows Explorer, it would start an IPython Notebook server and open in a browser right away. 

Something like this could be added to the registry during installation:
```
[HKEY_CLASSES_ROOT\Directory\shell\ipynb\command]
@="\"C:\\Anaconda\\Scripts\\ipython.exe\" \"notebook\" \"%V\""

[HKEY_CLASSES_ROOT\ipynb_auto_file\shell\open\command]
@="\"C:\\Anaconda\\Scripts\\ipython.exe\" \"notebook\" \"%1\""
```

**Comment by ccordoba12** [comment]
It's a good idea, but starting a new notebook server for every notebook seems overkill.

**Comment by petebachant** [comment]
True. I wasn't sure if there was a better way. Regardless of the solution, not being able to simply double-click a notebook to open it may be a turnoff to new users.

**Comment by mgaitan** [comment]
:+1: to tackle this

**Comment by psychemedia** [comment]
+1

We're running an open online course at the moment for programming novices, using Anaconda, and we're getting quite a few comments along the lines of "tried opening the ipynb file and Windows didn't recognise it". And yes, there are plenty of mentions in the instructions about making sure Anaconda is running and making sure learners open the notebooks from the Jupyter notebook homepage in their browser!;-)

(FWIW, we're also getting issues with the notebooks in IE - though not sure what (o/s, browser version ) combinations.)

**Comment by bollwyvl** [comment]
IE isn't officially supported by the notebook.

http://jupyter-notebook.readthedocs.org/en/latest/notebook.html#browser-compatibility

The safari is kind of a lie, as it is not automated, but there are enough
devs that work in osx that it's pretty robust.

Automated testing is the major issue, currently done on Travis. Perhaps an
appveyor pr could help towards this, though there have been several
discussion of moving to sauce labs. Mobile needs to be part of the story,
too, and they are about the only ones that hit them all.

I don't see the 4.x line adding ie as a supported target, but could be on
the roadmap for 5.x, if there was someone that championed it.

**Comment by ccordoba12** [comment]
IE 11 works ok on my (very limited) testing

**Comment by SimonBiggs** [comment]
Thomas Kluyver @takluyver has a very neat solution to this:
https://github.com/takluyver/nbopen

It only starts a new notebook server if one is not currently available for the requested directory.

**Comment by ijstokes** [comment]
@csoja having nbopen installed and configured by default as part of Anaconda installation would be amazing and would address some of the issues recently raised by @AllenDowney

**Comment by ccordoba12** [comment]
@ijstokes, we have an improved version of `nbopen` in the `spyder-notebook` plugin we are developing for Spyder.

So you should use our version in Navigator instead of the basic `nbopen` package (which only works in Python 3, by the way).

**Comment by ijstokes** [comment]
@ccordoba12 this won't be me implementing the association. In any case I think the default for .ipynb should be Jupyter ( or at this point JupyterLab), while .py should go to Spyder.  I believe this should be given as an implementation time option, defaulting to setting up the associations, with an option in navigator's configuration to remove or change the associations.

**Comment by ccordoba12** [comment]
> this won't be me implementing the association

Sure, I know that. By *you* I meant the Navigator team, sorry for the confusion.

> I think the default for .ipynb should be Jupyter ( or at this point JupyterLab), while .py should go to Spyder

Agreed too. I just wanted to mention that we have improved `nbopen` and that Navigator should use our improvements instead of the original project.

> with an option in navigator's configuration to remove or change the associations.

That would be a nice option to have.

**Comment by goanpeca** [comment]
This could be implemented on Navigator for sure. 

@chdoig, @pzwang, @fpliger thoughts?

**Comment by bilderbuchi** [comment]
> I just wanted to mention that we have improved nbopen and that Navigator should use our improvements instead of the original project.

Will those improvements make it back upstream?

**Comment by Ablinne** [comment]
How is the progress on this? Would be awesome if this would work out of the box. This is quite a showstopper when introducing Jupyter to newbies.
